types/sinon/index.d.ts

```diff
@@ -12,13 +12,19 @@
 
 import * as FakeTimers from "@sinonjs/fake-timers";
 
+type ObjectLike = Record<string | number | symbol, unknown>;
+
 // sinon uses DOM dependencies which are absent in browser-less environment like node.js
 // to avoid compiler errors this monkey patch is used
 // see more details in https://github.com/DefinitelyTyped/DefinitelyTyped/issues/11351
 interface Event {} // tslint:disable-line no-empty-interface
 interface Document {} // tslint:disable-line no-empty-interface
 
 declare namespace Sinon {
+    type DeepPartialOrMatcher<T> = {
+        [K in keyof T]?: SinonMatcher | (T[K] extends ObjectLike ? DeepPartialOrMatcher<T[K]> : T[K]);
```

Comment:
could we just assert that it extends `object` here like we do in `MatchArguments<T>`? rather than having to define `ObjectLike`

Reply:
Sure, I’ve copied this from another `DeepPartial` implementation within `DefinitelyTyped`. I have changed it now to `object`.